### ISSUE TITLE
Static Variable Initializers don't support LINQ

### ISSUE THREAD

**Issue by kenshinthebattosai** [bridgedotnet/Bridge#248]
In the case where:

````
public class Type {
    public String Name {get;set;}
    public Boolean IsPositive {get;set;}
    public static Type TypeA = new Type("A", false);
    public static Type TypeB = new Type("B");
    ...
    ...
    // Works
    public static readonly Type[] AllTypes = {
        TypeA,
        TypeB,
        ...
    };
    
    // Does not work -> Throws Build ERROR
    public static readonly Type[] PositiveTypes = AllTypes.Where(type=>type.IsPositive).ToArray();
}

**Comment by kenshinthebattosai** [comment]
A workaround at the moment is to use
````
public class Type {
    ...
    ...
    private static Type[] _positiveTypes;
    public static Type[] PositiveTypes
    {
        get
        {
            return _positiveTypes ??
                (_positiveTypes = AllTypes.Where(type=>type.IsPositive).ToArray());
        }
    }
    ...
    ...
}
````

**Comment by fabriciomurta** [comment]
Hello Battosai! :)

Thank you for sharing your problem with Bridge.NET. We'll try our best to help you!

For the sample you provided I couldn't make your last line work on an ordinary C# program as well.
I've created a plain C# ClassLibrary and pasted your class, completing with the needed constructor, so that I have this code on the file:
```csharp
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ClassLibrary3
{
    public class Class1
    {
    }

    public class Type
    {
        public String Name { get; set; }
        public Boolean IsPositive { get; set; }
        public static Type TypeA = new Type("A", false);
        public static Type TypeB = new Type("B");

        public Type(String character, Boolean trueState = true)
        {
            Name = character;
            IsPositive = trueState;
        }

        // Works
        public static readonly Type[] AllTypes = {
            TypeA,
            TypeB
        };

        // Does not work -> Throws Build ERROR
        public static readonly Type[] PositiveTypes = AllTypes.Where(type => type.IsPositive);
    }
}
```

What I get (even before trying to build) is in the screenshot below:
![Error message](https://cloud.githubusercontent.com/assets/10730372/7550814/bd5609d8-f646-11e4-83d6-1375327fe122.png)

Again, this is in an ordinary C# program, not Bridge. This result suggests this construct is not supported by linq at all.

**Comment by kenshinthebattosai** [comment]
.ToArray() at the end.

**Comment by fabriciomurta** [comment]
Alright, I didn't notice the change.

**Comment by fabriciomurta** [comment]
Alright, I have confidence that this issue is now fixed on our git repository. The fix will go live in version 1.5. Thanks again for reporting it!
